File: OneShotOneKill_26.1.2/src/main/java/de/oneshotonekill/manager/KillEffectManager.java
package de.oneshotonekill.manager;

import net.kyori.adventure.text.Component;
import net.kyori.adventure.text.minimessage.MiniMessage;
import org.bukkit.Color;
import org.bukkit.Location;
import org.bukkit.Particle;
import org.bukkit.Sound;
import org.bukkit.SoundCategory;
import org.bukkit.entity.Player;

import java.util.HashMap;
import java.util.Map;
import java.util.UUID;

public class KillEffectManager {

    public enum KillEffect {
        LIGHTNING("<yellow>[⚡] Blitzschlag</yellow>"),
        FIREWORK("<red>[🎆] Feuerwerks-Explosion</red>"),
        BLOOD("<dark_red>[🩸] Blut-Splash</dark_red>"),
        ENDER("<light_purple>[🔮] Ender-Portal</light_purple>"),
        TOTEM("<green>[✨] Totem-Aura</green>"),
        NONE("<gray>[❌] Keine Animation</gray>");

        private final Component displayName;

        KillEffect(String miniMessageFormat) {
            this.displayName = MiniMessage.miniMessage().deserialize(miniMessageFormat);
        }

        public Component getDisplayName() {
            return displayName;
        }
    }

    private final Map<UUID, KillEffect> selectedEffects = new HashMap<>();

    public KillEffect getSelectedEffect(UUID uuid) {
        return selectedEffects.getOrDefault(uuid, KillEffect.LIGHTNING);
    }

    public void setSelectedEffect(UUID uuid, KillEffect effect) {
        selectedEffects.put(uuid, effect);
    }

    public void playKillEffect(Player killer, Location deathLoc) {
        KillEffect effect = getSelectedEffect(killer.getUniqueId());
        if (deathLoc == null || deathLoc.getWorld() == null) return;

        switch (effect) {
            case LIGHTNING -> {
                deathLoc.getWorld().strikeLightningEffect(deathLoc);
            }
            case FIREWORK -> {
                deathLoc.getWorld().spawnParticle(Particle.FIREWORK, deathLoc.clone().add(0, 1, 0), 40, 0.4, 0.4, 0.4, 0.1);
                deathLoc.getWorld().playSound(deathLoc, Sound.ENTITY_FIREWORK_ROCKET_BLAST, SoundCategory.MASTER, 1.0f, 1.0f);
            }
            case BLOOD -> {
                Particle.DustOptions redDust = new Particle.DustOptions(Color.RED, 1.8f);
                deathLoc.getWorld().spawnParticle(Particle.DUST, deathLoc.clone().add(0, 1, 0), 60, 0.4, 0.5, 0.4, 0.1, redDust);
                deathLoc.getWorld().playSound(deathLoc, Sound.ENTITY_SQUID_SQUIRT, SoundCategory.MASTER, 1.0f, 0.8f);
            }
            case ENDER -> {
                deathLoc.getWorld().spawnParticle(Particle.DRAGON_BREATH, deathLoc.clone().add(0, 1, 0), 40, 0.4, 0.4, 0.4, 0.05);
                deathLoc.getWorld().spawnParticle(Particle.PORTAL, deathLoc.clone().add(0, 1, 0), 50, 0.5, 0.5, 0.5, 0.2);
                deathLoc.getWorld().playSound(deathLoc, Sound.ENTITY_ENDERMAN_TELEPORT, SoundCategory.MASTER, 1.0f, 0.6f);
            }
            case TOTEM -> {
                deathLoc.getWorld().spawnParticle(Particle.TOTEM_OF_UNDYING, deathLoc.clone().add(0, 1, 0), 60, 0.4, 0.5, 0.4, 0.2);
                deathLoc.getWorld().playSound(deathLoc, Sound.ITEM_TOTEM_USE, SoundCategory.MASTER, 0.8f, 1.2f);
            }
            case NONE -> {
                // Keine Spezial-Animation
            }
        }
    }
}

File: src/main/kotlin/de/oneshotonekill/manager/KillEffectManager.kt
package de.oneshotonekill.manager

import net.kyori.adventure.sound.Sound
import org.bukkit.Color
import org.bukkit.Location
import org.bukkit.Material
import org.bukkit.Particle
import org.bukkit.Sound as BukkitSound

class KillEffectManager {

    /**
     * Spielt nativ ueber die Paper Particle & Sound Engine den Blut-Splash-Killeffekt an der
     * Position ab.
     */
    fun playKillEffect(deathLoc: Location?) {
        val world = deathLoc?.world ?: return
        val particleLoc = deathLoc.clone().add(0.0, 1.0, 0.0)

        // 1. Rote Staub-Partikel fuer dichten Blutspritzer
        val redDust = Particle.DustOptions(Color.fromRGB(180, 0, 0), 2.2f)
        world.spawnParticle(Particle.DUST, particleLoc, 75, 0.4, 0.6, 0.4, 0.1, redDust)

        // 2. Redstone-Block Partikel fuer spritzende Blutpartikel
        world.spawnParticle(
            Particle.BLOCK, particleLoc, 50, 0.3, 0.5, 0.3, 0.1,
            Material.REDSTONE_BLOCK.createBlockData(),
        )

        // 3. Native Audio-Feedback
        world.playSound(
            Sound.sound(BukkitSound.ENTITY_SQUID_SQUIRT, Sound.Source.MASTER, 1.0f, 0.7f),
            deathLoc.x(), deathLoc.y(), deathLoc.z(),
        )
        world.playSound(
            Sound.sound(BukkitSound.ENTITY_PLAYER_HURT, Sound.Source.MASTER, 0.8f, 0.6f),
            deathLoc.x(), deathLoc.y(), deathLoc.z(),
        )
    }
}
